src/component.coffee

```diff
@@ -381,55 +374,61 @@ Ember.AddeparMixins.ResizeHandlerMixin,
   # TODO: Make private or reorganize into a new section
   # ---------------------------------------------------------------------------
 
+  lastSelected: null
+
   isSelected: (row) ->
     @get('selection').contains row.get('content')
 
   setSelected: (row, val) ->
     @persistSelection()
+    item = row.get 'content'
     if val
-      @get('persistedSelection').addObject row
+      @get('persistedSelection').addObject item
     else
-      @get('persistedSelection').removeObject row
+      @get('persistedSelection').removeObject item
 
-  # rows that were selected directly or as part of a previous
+  # items that were selected directly or as part of a previous
   # range selection (shift-click)
   persistedSelection: Ember.computed -> Ember.A()
 
-  # rows that are part of the currently editable range selection
+  # items that are part of the currently editable range selection
   rangeSelection: Ember.computed -> Ember.A()
 
-  _selection: Ember.computed ->
-    @get('persistedSelection').copy().addObjects(@get('rangeSelection'))
-  .property 'persistedSelection.[]', 'rangeSelection.[]'
-
+  # TODO: Handle click event in the row view
   click: (event) ->
     row = @getRowForEvent event
-    return unless row
+    item = row?.get 'content'
+    return unless item
     return if @get('selectionMode') is 'none'
     if @get('selectionMode') is 'single'
       @get('persistedSelection').clear()
-      @get('persistedSelection').addObject row
+      @get('persistedSelection').addObject item
     else
       if event.shiftKey
         @get('rangeSelection').clear()
 
         lastIndex = @rowIndex(@get('lastSelected'))
+        # If the last selected row is no longer in the table, use the
+        # first row in the table
+        lastIndex = 0 if lastIndex is -1
```

Comment:
just questioned this, checked against finder, and finder does the same thing as the code above. :+1: